fe/fe-core/src/main/java/org/apache/doris/datasource/credentials/CredentialUtils.java

```diff
@@ -17,22 +17,69 @@
 
 package org.apache.doris.datasource.credentials;
 
-import com.google.common.annotations.VisibleForTesting;
+import org.apache.doris.datasource.property.storage.StorageProperties;
 
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
 
+/**
+ * Utility class for Credential operations
+ */
 public class CredentialUtils {
+
+    /**
+     * Supported cloud storage prefixes for filtering vended credentials
+     */
+    private static final Set<String> CLOUD_STORAGE_PREFIXES = new HashSet<>(Arrays.asList(
+            "s3.",      // Amazon S3
+            "oss.",     // Alibaba OSS
+            "cos.",     // Tencent COS
+            "obs.",     // Huawei OBS
+            "gs.",      // Google Cloud Storage
+            "azure."    // Microsoft Azure
+    ));
+
+    /**
+     * Filter cloud storage properties from raw vended credentials
+     * Only keeps properties with supported cloud storage prefixes
+     *
+     * @param rawVendedCredentials Raw vended credentials map
+     * @return Filtered cloud storage properties
+     */
+    public static Map<String, String> filterCloudStorageProperties(Map<String, String> rawVendedCredentials) {
```

Comment:
There’s no guarantee that all properties we need start with these prefixes. If new ones are introduced, the code will need to be updated as well.